src/core/crypto/crux.rs

```diff
@@ -151,3 +151,40 @@ impl DataDecrypt<DecryptedDataGroup> for EncryptedDataGroup {
             )?))
     }
 }
+
+#[async_trait::async_trait]
+impl DataEncrypt<EncryptedData> for DecryptedData {
+    async fn encrypt(
+        self,
+        state: &AppState,
+        identifier: &Identifier,
+    ) -> errors::CustomResult<EncryptedData, errors::CryptoError> {
+        let version = Version::get_latest(identifier, state).await;
+        let decrypted_key = Key::get_key(state, identifier, version).await.switch()?;
```

Comment:
Try to avoid passing whole state, if the child function uses only a part of it.

Reply:
Resolving, Since we are using KeyManagerClient in `get_key` as well